content/docs/specs/native_histograms.md

```diff
@@ -0,0 +1,2353 @@
+
+While partitioning of a classic histogram with many buckets by labels has to be
+done judiciously, the situation is more relaxed with native histograms.
+Partitioning a native histograms still creates a multiplicity of individual
+histograms. However, the resulting partitioned histograms will often populate
+fewer buckets each than the original unpartitioned histogram. (For example, if
+a histogram tracking the duration of HTTP requests is partitioned by HTTP
+status code, the individual histogram tracking requests responded by status
+code 404 might have a very sharp bucket distribution around the typical
+duration it takes to identify an unknown path, populating only a few buckets.)
+The total number of populated buckets for all partitioned histograms will still
+go up, but by a smaller factor than the number of partitioned histograms. (For
+example, if adding labels to an already quite heavy classic histogram results
+in 100 labeled histograms, the total cost will go up by a factor of 100. In
+case of a native histogram, the cost for the single histogram might already be
+lower if the classic histogram featured a high resolution. After partitioning,
+the total number of populated buckets in the labeled native histograms will be
+signifcantly smaller than 100 times the number of buckets in the original
+native histogram.)
+
+### NHCB
+
+Currently (2024-11-03), instrumentation libraries offer no way to directly
+configure native histograms with custom bucket boundaries (NHCBs). The use case
+for NHCBs is to allow native-histogram enabled scrapers to convert classic
+histograms to NHCBs upon ingestion (see [next section](#scrape-configuration)).
+However, there are valid use cases where custom buckets are desirable directly
+during instrumentation. In those cases, the current approach is to instrument
+with a classic histogram and configure the scraper to convert it to an NHCB
+upon ingestion. However, a more direct treatment of NHCBs in instrumentation
+libraries might happen in the future.
+
+## Scrape configuration
+
+To enable the Prometheus server to scrape native histograms, the feature flag
+`--enable-feature=native-histograms` is required. This flag also changes the
+content negotiation to prefer the classic protobuf-based exposition format over
+the OpenMetrics text format. (TODO: This behavior will change once native
+histograms are a stable feature.)
+
+### Finetuning content negotiation
+
+With Prometheus v2.49 and later, it is possible to finetune the scrape protocol
+negotiation globally or per scrape config via the `scrape_protocols` config
+setting. It is a list defining the content negotiation priorities. Its default
+value depends on the `--enable-feature=native-histograms` flag. If the flage is
+set, it is `[ PrometheusProto, OpenMetricsText1.0.0, OpenMetricsText0.0.1,
+PrometheusText0.0.4 ]`, otherwise `[ OpenMetricsText1.0.0,
+OpenMetricsText0.0.1, PrometheusText0.0.4 ]`. This results in the behavior
```

Comment:
Should the diff between the lists be explicitly pointed out - i.e. point out the difference is `PrometheusProto`. Just so you don't need to jump between the lists to figure out the change.